From 43c6174d6ba106c3bd5acabb92a008acbcb64e62 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 04 Jul 2024 15:48:08 +0300
Subject: [PATCH] [FSW-2704] fs1: add fake fuse and rma

This patch adds empty functions for fake fuse init
and rma checks.

Change-Id: I7679a13daf514105643955afbfc33d850263fe5f
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index e49551d..5198e4e 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -35,6 +35,10 @@
 
 	boot_rom_tpm_spi_setup();
 
+	boot_rom_fake_fuse();
+
+	boot_rom_check_rma_enabled();
+
 	boot_rom_security_init();
 
 	boot_rom_wait_for_signal();
diff --git a/common/common.c b/common/common.c
index 3fe2322..7f635ec 100644
--- a/common/common.c
+++ b/common/common.c
@@ -89,8 +89,6 @@
 
 	boot_rom_eom_overrides();
 
-	/* TODO: Enable FAKE FUSE & RMA support */
-
 	boot_rom_init_sku();
 
 	return;
@@ -107,6 +105,16 @@
 	return p_img_ptr_ctx->boot_desc_flag;
 }
 
+void boot_rom_fake_fuse(void)
+{
+	/* Empty Body */
+}
+
+void boot_rom_check_rma_enabled(void)
+{
+	/* Empty Body */
+}
+
 void boot_rom_check_n_enter_recovery(void)
 {
 	bool override = true;
diff --git a/include/boot_rom.h b/include/boot_rom.h
index b2f48a3..bc4e084 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -109,6 +109,8 @@
 uint8_t boot_rom_get_die_type(void);
 void boot_rom_init(void);
 void boot_rom_fuse(void);
+void boot_rom_fake_fuse(void);
+void boot_rom_check_rma_enabled(void);
 void boot_rom_wait_for_signal(void);
 void boot_rom_check_n_enter_recovery(void);
 void boot_rom_bringup_ppboot(void);
